trojsten/settings/production.py

```diff
@@ -1,28 +1,19 @@
+import warnings
+
 from django.core.exceptions import ImproperlyConfigured
 
 from trojsten.settings.common import *
 
 
 def requiredenv(name):
     if name not in os.environ:
-        raise ImproperlyConfigured("Value %s missing in environment configuration" % name)
+        warnings.warn("Value %s missing in environment configuration" % name)
```

Comment:
why this change? this function is wrapping environment variables, which can cause errors or application crashes if missing in production and it's purpose is to make the app crash with a nicer warning message at startup, so that we know something was misconfigured during deployment, not at the moment when someone tries to login with github.

when this is used only in production environments, i don't see how it can be properly used with just issuing a warning instead of crashing directly.

Reply:
This is read with every ./manage.py command, including those that don't need all the settings (e.g. compilemessages). If we don't have the particular env set during docker build stage the command would crash.
We set the oauth keys only in runtime, so this is a workaround.

Another options would be to have fake env in the build stage, but that would defeat the purpose of required env (the app would still not work with wrong data), or I can try to detect if this is used from some ./manage.py command and not crash in that case.
What would you prefer?

Reply:
I usually run compilemessages and collectstatic with development/debug configuration and everything usually works fine.

Reply:
We use different storage backend in production settings, so collectstatic would behave differently.

Reply:
oh 😄 i didn't even think about this, because i usually use whitenoise as a static server which can handle running in development with production settings